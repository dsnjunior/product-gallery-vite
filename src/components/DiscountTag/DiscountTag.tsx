import { Typography } from '../Typography';

export type DiscountTagProps = {
  discount: number;
};

export function DiscountTag({ discount }: DiscountTagProps) {
  return (
    <Typography
      variant="c100"
      weight="500"
      component="span"
      className="border shadow-border border-current rounded text-neutral-700 px-2 h-6 inline-flex items-center"
    >
      {discount}% Off
    </Typography>
  );
}
