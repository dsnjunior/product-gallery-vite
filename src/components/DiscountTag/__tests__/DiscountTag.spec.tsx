import { render, screen } from '@testing-library/react';
import { describe, it } from 'vitest';

import { DiscountTag } from '../DiscountTag';

describe('DiscountTag.tsx', () => {
  it('should render discount tag', () => {
    render(<DiscountTag discount={35} />);

    const discountTag = screen.getByText('35% Off');

    expect(discountTag).toBeInTheDocument();

    expect(discountTag).toHaveClass(
      'border shadow-border border-current rounded text-neutral-700 px-2 h-6 inline-flex items-center',
    );

    expect(discountTag).toMatchSnapshot();
  });

  it('should render discount tag with 0% discount', () => {
    render(<DiscountTag discount={0} />);

    const discountTag = screen.getByText('0% Off');

    expect(discountTag).toBeInTheDocument();
  });

  it('should render discount tag with 100% discount', () => {
    render(<DiscountTag discount={100} />);

    const discountTag = screen.getByText('100% Off');

    expect(discountTag).toBeInTheDocument();
  });
});
