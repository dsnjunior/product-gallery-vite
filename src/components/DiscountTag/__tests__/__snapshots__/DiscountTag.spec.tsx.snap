// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DiscountTag.tsx > should render discount tag 1`] = `
<span
  class="text-c100 font-500 border shadow-border border-current rounded text-neutral-700 px-2 h-6 inline-flex items-center"
>
  35
  % Off
</span>
`;
